File: Byte&Buy.Backend/src/ByteBuy.Core/DTO/Offer/SaleOffer/SaleOfferDetailsResponse.cs
using ByteBuy.Core.DTO.Image;
using ByteBuy.Core.DTO.Money;

namespace ByteBuy.Core.DTO.Offer.SaleOffer;

public record SaleOfferDetailsResponse(
    Guid Id,
    int QuantityAvaliable,
    MoneyDto PricePerItem,
    string Condition,
    string Category,
    string Description,
    string Title,
    Object Seller,
    IReadOnlyCollection<ImageResponse> Images);

File: Byte&Buy.AngularSPA/src/app/core/dto/offers/sale/sale-offer-details-response.ts
import { Guid } from "guid-typescript";
import { MoneyDto } from "../../common/money-dto";
import { CompanySellerResponse } from "../common/company-seller-response";
import { PrivateSellerResponse } from "../common/private-seller-response";
import { ImageResponse } from "../../image/image-response";
import { OfferStatus } from "../enum/offer-status";

export interface SaleOfferDetailsResponse {
    id: Guid;
    quantityAvaliable: number;
    status: OfferStatus;
    pricePerItem: MoneyDto;
    condition: string;
    category: string;
    description: string;
    title: string;
    seller: CompanySellerResponse | PrivateSellerResponse;
    images: ImageResponse[];
}